src/particles/plasma/PlasmaParticleContainer.cpp

```diff
@@ -329,26 +334,31 @@ IonizationModule (const int lev,
 
         long num_ions = ptile_ion.numParticles();
 
+
+        // This kernel supports multiple deposition orders (0, 1, 2, 3) at compile time
+        // and calculates ionization probability. If ionization occurs, it increments
+        // `p_num_new_electrons` to calculate the number of ionized electrons.
```

Comment:
```suggestion
        // `p_num_new_electrons` to calculate the number of ionized electrons.
        // It also constructs a mask with 1 boolean per macro-ion: 1 if ionized, 0 otherwise. 
```